SignalServiceKit/src/Loki/Protocol/Session Management/LokiSessionResetImplementation.swift

```diff
@@ -44,19 +45,29 @@ public class LokiSessionResetImplementation : NSObject, SessionResetProtocol {
             return
         }
         guard !recipientID.isEmpty else { return }
+        let masterHexEncodedPublicKey = storage.getMasterHexEncodedPublicKey(for: recipientID, in: transaction) ?? recipientID
         guard let thread = TSContactThread.getWithContactId(recipientID, transaction: transaction) else {
             Logger.debug("[Loki] A new session was adopted but the thread couldn't be found for: \(recipientID).")
             return
         }
         // If the current user initiated the reset then send back an ephemeral message to acknowledge the completion of the session reset
-        if thread.sessionResetStatus == .initiated {
-            SessionManagementProtocol.sendSessionEstablishedMessage(to: recipientID, in: transaction)
+        if let masterThread = TSContactThread.getWithContactId(masterHexEncodedPublicKey, transaction: transaction), masterThread.sessionResetStatus == .initiated {
```

Comment:
What about cases where you also want to reset the session with a secondary device? This is probably the direction we're heading in the future where triggering a session reset should send out the message to all linked devices